flycheck.el

```diff
@@ -4214,6 +4214,20 @@ universal prefix arg, and only the id with normal prefix arg."
     (`(eval ,_) t)
     (_ nil)))
 
+(defun flycheck-cwd-wrapper (checker)
+  "Function returning apropriate current working directory.
+
+Work out the working directory from which the CHECKER command
+will be called.  The default is the `default-directory' of the source-buffer
+from which flycheck is calling the checker.
+:cwd can be a list that contains a fixed path string or a function that will
+return a path.  The function needs to return an absolut path not a relative path."
+  (let ((cwd (flycheck-checker-get checker 'cwd)))
+    (or (and (functionp cwd) (funcall cwd))
+        (and (listp cwd) (stringp (car cwd)) (file-truename (car cwd)))
```

Comment:
Would you mind to explain the purpose of this line?  As far as I can see this expects a single element list containing a string.  But why a list, and not just a plain string right away? 

Also, what use case does a constant string as working directory address?